src/types/union.md

```diff
@@ -24,5 +24,36 @@ The memory layout of a `union` is undefined by default (in particular, fields do
 *not* have to be at offset 0), but the `#[repr(...)]` attribute can be used to
 fix a layout.
 
+## Union Values
+
+r[type.union.value]
+
+r[type.union.value.value-bytes]
+A byte `b` in the representation of a struct or union is a value byte if there exists a field of that aggregate such that:
+* The field has some type `T`,
+* The offset of that field `o` is such that `b` falls at an offset in `o..(o+size_of::<T>())`,
+* Either `T` is a primitive type or the offset of `b` within the field is not a padding byte in the representation of `T`.
```

Comment:
Couldn’t this be simplified to the latter condition only if we specify that primitive types have no padding bytes?